libc-bottom-half/crt/crt1-command.c

```diff
@@ -7,6 +8,11 @@ extern void __wasm_call_dtors(void);
 // that the `_start` function isn't started more than once.
 static volatile int started = 0;
 
+static void dummy_0()
+{
+}
+weak_alias(dummy_0, __wasi_init_tp);
```

Comment:
Instead of using a weak symbol, could you put this under `#if defined(_REENTRANT)` so that it isn't included in non-threaded builds?